app/interactors/create_user.rb

```diff
@@ -0,0 +1,25 @@
+class CreateUser
+  include Interactor
+
+  def call
+    validate_input
+    execute
+    validate_output
+  end
+
+  private
+
+  def validate_input
+    context.fail!(error: "invalid user params") unless context.user_params
+  end
+
+  def execute
+    context.user = User.create!(context.user_params)
+  rescue
```

Comment:
I would say normally it is bad practice, but I've had this burning in the back of my head for a while :smile:. I certainly have tried to do something like this:

```ruby
def call
  validate_input
  execute
  validate_output
rescue
  # should probably log the exception to keep track of it
  context.fail!(error: :internal_server_error)
end
```

The idea is so that the API can fail gracefully. We can handle the internal server error however we want. Showing a message, for instance.

The problem comes from obscuring the failure. It makes it harder to find the real culprit if you don't log this somewhere to keep track of bugs. It's also very important to first handle the possible exceptions that could happen in your code before having a "catch-all" net. For example `ActiveRecord::RecordNotFound`. You should handle that before hand, and display the correct error: `not_found` instead of `internal_server_error`. This comes back to expressing exactly what went wrong to a developer using your API.